state/processor.go

```diff
@@ -64,6 +67,34 @@ func NewTransactionProcessor(rnd PseudoRandomizer, db *StateDB, gasParams GasCon
 	}
 }
 
+func PublicKeyToAccountAddress(pub ed25519.PublicKey) address.Address {
+	var addr address.Address
+	addr.SetBytes(pub[:common.AddressLength])
+	return addr
+}
+
+// Validate the tx's signature by extracting the source account and validating its existence.
+// Return the src acount address and error in case of failure
+func (tp *TransactionProcessor) ValidateTransactionSignature(tx types.SerializableSignedTransaction) (address.Address, error) {
```

Comment:
rename func to something like ValidateExtractedPublicKey

Reply:
why? it is not general, it is specific for txs

Reply:
IMO the name does not express what the function does sinceit first extracts public key and validates it. these are two seperate things that also IMO shouldn't be combined in one function